File: tools/benchmark.py
import gc
import sys
import time
import pandas

filename = sys.argv[1]
if len(sys.argv) > 2:
    params = eval(sys.argv[2])
else:
    params = {}

print("package,run,elapsed")
for i in range(6):
    gc.collect()
    start = time.time()
    pandas.read_csv(filename, **params)
    elapsed = time.time() - start
    print("pandas", ",", i + 1, ",", elapsed, sep="")

File: tools/benchmark.jl
using TableReader

filename = ARGS[1]
if length(ARGS) > 1
    params = eval(Meta.parse(ARGS[2]))
    @assert params isa NamedTuple
else
    params = ()
end

println("package,run,elapsed,gctime,bytes")
for i in 1:6
    GC.gc(); GC.gc()
    _, elapsed, bytes, gctime, _ = @timed readcsv(filename; params...)
    println("TableReader.jl", ',', i, ',', elapsed, ',', gctime, ',', bytes)
end
